cns/restserver/util.go

```diff
@@ -441,6 +499,45 @@ func (service *HTTPRestService) SendNCSnapShotPeriodically(ncSnapshotIntervalInM
 	}
 }
 
+// isNCWaitingForUpdate :- Determine whether NC version on NMA matches programmed version
+func isNCWaitingForUpdate(ncVersion, ncid string) (waitingForUpdate bool, returnCode int, message string) {
+	getNCVersionURL, ok := ncVersionURLs.Load(ncid)
+	if !ok {
+		returnCode = NotFound
+		message = fmt.Sprintf("[Azure-CNS] Network container %s not found", ncid)
+		return
+	}
+
+	response, err := nmagentclient.GetNetworkContainerVersion(ncid, getNCVersionURL.(string))
+	if err == nil {
+		if response.StatusCode == http.StatusOK {
+			var versionResponse nmagentclient.NetworkContainerResponse
+			rBytes, _ := ioutil.ReadAll(response.Body)
+			json.Unmarshal(rBytes, &versionResponse)
+			if versionResponse.ResponseCode == "200" {
+				programmedVersion, _ := strconv.Atoi(ncVersion)
+				nmaVersion, _ := strconv.Atoi(versionResponse.Version)
+				if programmedVersion > nmaVersion {
+					waitingForUpdate = true
+					returnCode = NetworkContainerPendingStatePropagation
+					message = fmt.Sprintf("[Azure-CNS] Network container %s v%d had not propagated to respective NMA w/ v%d", ncid, programmedVersion, nmaVersion)
+				}
```

Comment:
do we need to log if programmedversion < nmaversion?

Reply:
no, CNS should be receiving latest